### PR TITLE
Challenge3

### PR DESCRIPTION
**Pull request by OfekPoraz**

### CODE REVIEW

File: src/com/company/three/FileEncryptorTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
```
Comment (ido9000) [inline src/com/company/three/FileEncryptorTest.java:14]: Should be in test directory not source directory

File: src/com/company/three/FileEncryptorTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        for (FileEncryptor FETLoop : FET){
+            encryptedPath.add(FETLoop.createEncryptedDecryptedFile(originalPath, true, FETLoop.getName()));
+            keyPath.add(FETLoop.setNewKey(originalPath, FETLoop.getName()));
+            FETLoop.encryptFile(originalPath,encryptedPath.get(i), keyPath.get(i));
+            decryptedPath.add(FETLoop.createEncryptedDecryptedFile(originalPath, false, FETLoop.getName()));
+            i++;
+
+        }
+    }
+
```
Comment (ido9000) [inline src/com/company/three/FileEncryptorTest.java:40]: assume the test failed, who do you know which algorithm failed in test?

Hint: you don't know! 
please create separate tests for each algorithm.
also, check edge cases (file not found, null pointer etc)

File: src/com/company/three/FileEncryptorTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    @BeforeAll
+    public static void beforeAll() throws IOException {
+        FET.add(new FileEncryptor(new ShiftUpEncryption()));
+        FET.add(new FileEncryptor(new ShiftMultiplyEncryption()));
+        FET.add(new FileEncryptor(new DoubleEncryption(new ShiftMultiplyEncryption(), new ShiftUpEncryption())));
+        FET.add(new FileEncryptor(new RepeatEncryption(4, new ShiftUpEncryption())));
+        FET.add(new FileEncryptor(new XorEncryption()));
+        int i = 0;
+        for (FileEncryptor FETLoop : FET){
```
Comment (ido9000) [inline src/com/company/three/FileEncryptorTest.java:31]: don't check who encrypt algorithm work in this function.
check the encrypt algorithm in separate tests for each algorithm
include edge cases

**Review by ido9000 — CHANGES_REQUESTED** [review]
